cipd_packages/device_doctor/lib/src/android_device.dart

```diff
@@ -380,7 +381,68 @@ class AndroidDevice implements Device {
 
   @override
   Future<void> recover() async {
-    await eval('adb', <String>['-s', deviceId!, 'reboot'], canFail: false);
+    await cleanDevice();
+  }
+
+  @visibleForTesting
+  Future<void> deletePackageCache() async {
+    final ProcessResult result = Process.runSync('adb', <String>['shell', 'pm', 'list', 'packages']);
+
+    if (result.exitCode != 0) {
+      throw Exception(result.stderr as String);
+    }
+    final packages = <String>[];
+
+    // Listen to the stdout and stderr streams
+    await result.stdout.transform(utf8.decoder).forEach((data) {
```

Comment:
you shouldn't need to do this. unless you override the encoding, the default type of `result.stdout` should be a string. from there you can do `(result.stdout as String).split('\n').forEach((data) {`